proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,99 @@
+# **Add `search` Functionality to DotNet**
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- [Nigusu](https://github.com/Nigusu-Allehu) <!-- GitHub username link -->
+- [GitHub Issue](https://github.com/NuGet/Home/issues/6060) <!-- GitHub Issue link -->
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+The specification seeks to add search functionality to dotnet. The functionality will be added to dotnet as a search command: `dotnet package search`.  
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+The `dotnet` CLI is missing a way for customers to search for packages from the command line, as is possible with `nuget.exe search`.
+This brings NuGet experiences in the `dotnet` CLI one step closer towards feature parity.
+
+While [NuGet/Home#6060](https://github.com/NuGet/Home/issues/6060) (the linked issue) also discusses the `nuget.exe list` command, the issue was created before `nuget.exe search` was added.
+The [NuGet Server API Search Resource specification requires all packages to be returned when no `q` parameter is provided](https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource#request-parameters).
+Therefore, this proposal does not add a `list` equivalent to the `dotnet` CLI, as it is effectively redundant.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+Imagine you have your dotnet CLI open. You would like to look up a NuGet package named `MyPackage` from a source `<MySource>`. No worries, you can just use the following command : `dotnet package search MyPackage --source <MySource>`. It will provide you with the table of the packages in the source `<MySource>` that match with the search criteria. This output will have the search term used highlighted in the search result.
+
+The `package search [search terms] [options]` command will have the following options
+
+| Option | Function |
+|---------|:----------|
+| `--source` | A source to search |
+| `--exact-match` | Return exact matches only as a search result |
+| `--prerelease` | Allow prerelease packages to be shown. |
+| `--interactive` | Allows the command to stop and wait for user input or action (for example to complete authentication).|
+| `--take` | The number of results to return. The default value is 20.|
+| `--skip` | The number of results to skip, for pagination. The default value is 0. |
```

Comment:
I am reviewing the feature and have a question regarding the pagination aspect, especially when both --skip and --take options are used together.

From my understanding, introducing these options implies that users might view the search results in a paginated manner. If a user passes both the `--skip` and `--take` options, would the command be designed to wait for user input after displaying the first page of results? This could offer users an opportunity to view subsequent pages without having to re-run the command, enhancing user experience.

Reply:
The `--skip` and `--take` options are passed to the search api as parameters. By using `--skip x`, users will be able to skip the first x packages from being returned in the search result. By using `--take x`, users would be able to limit the maximum number of packages outputted at once. https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource